crawl-ref/source/chardump.cc

```diff
@@ -222,12 +346,59 @@ static void _sdump_header(dump_params &par)
     }
 }
 
+static void _json_sdump_header(json_dump_params &jpar)
+{
+    JsonNode *header(json_mkobject());
+
+    // Game type
+    string type = crawl_state.game_type_name();
+    if (type.empty())
+        type = CRAWL;
+    else
+        type += " DCSS";
+    json_append_member(header, "type", json_mkstring(type.c_str()));
+
+    // Version
+    json_append_member(header, "version", json_mkstring(Version::Long));
+
+    // Client
+    string client = "console";
+#ifdef USE_TILE_LOCAL
+    client = "tiles";
+#elif defined(USE_TILE_WEB)
+    if (::tiles.is_controlled_from_web())
+        client = "webtiles";
+    else
+        client = "console";
+#else
+    client = "console";
+#endif
+    json_append_member(header, "client", json_mkstring(client.c_str()));
+
+    // Seed
+    if (you.game_is_seeded
+#ifdef DGAMELAUNCH
+        && jpar.se // for online games, only show seed for a dead char
+#endif
+        )
+    {
+        json_append_member(header, "seed", json_mknumber(crawl_state.seed));
```

Comment:
This should be stored as a string, otherwise large seeds could lose precision when stored as a floating point number.